reference.py

```diff
@@ -51,8 +51,7 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
     if is_p2sh(vin.prevout):
         redeem_script = vin.scriptSig[1:]
         if is_p2wpkh(redeem_script):
-            txin = CTxInWitness().deserialize(redeem_script)
-            pubkey = ECPubKey().set(txin.scriptWitness.stack[-1])
+            pubkey = ECPubKey().set(vin.txinwitness.scriptWitness.stack[-1])
```

Comment:
since we are in a P2SH context, it should still be `CTxInWitness.deserialize(redeem_script)`. I was confused as to how you were getting the pubkey from the `txinwitness`, since this should be empty.

The witness field is always empty for P2SH spends because we are getting the witness from inside the redeem script.

Reply:
@josibake I thought this too but P2SH(P2WPKH) addresses need the witness stack. I tested this with my node

Reply:
Oh wow, TIL! I don't think I ever realized this before. Ignore my last few comments, what you have is correct :sweat_smile: